src/app/serve.rs

```diff
@@ -1,44 +1,43 @@
-use crate::core::listen::{Accept, Listen};
-use crate::{drain, task, Error};
+use super::accept_error::AcceptError;
+use crate::core::listen::{Accept, Listen, Serve};
+use crate::{drain, task};
 use futures::{try_ready, Future, Poll};
-use std::net::SocketAddr;
-use tokio::net::TcpStream;
-use tracing::{debug, info_span};
+use tracing::info_span;
 use tracing_futures::Instrument;
 
 /// Spawns a task that binds an `S`-typed server with an `L`-typed listener until
 /// a drain is signaled.
-pub fn serve<L, A>(
-    server: &'static str,
-    listen: L,
-    accept: A,
-    drain: drain::Watch,
-) -> impl Future<Item = (), Error = ()>
+pub fn spawn<L, A>(server: &'static str, listen: L, accept: A, drain: drain::Watch)
```

Comment:
nit: I think documentation is old. There is no longer an `S`, and `L` and `A` are now linked... `A` accepts `L::Connection`-typed connections.